### PR TITLE
Bumps Gateway API Dependencies

### PR DESCRIPTION
**Pull request by danehans**
Bumps Gateway API dependencies from v1.2.0 to v1.2.1.
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/10456

### CODE REVIEW

**Review by jenshu — APPROVED** [review]

File: .github/workflows/.env/nightly-tests/max_versions.env [hunk 1]
```diff
@@ -4,4 +4,4 @@ kind_version='v0.24.0'
 helm_version='v3.14.4'
 argocd_version='v2.8.4'
 istio_version='1.22.0'
-k8sgateway_api_version='v1.2.0'
+k8sgateway_api_version='v1.2.1'
```
Comment (sam-heilbron) [inline .github/workflows/.env/nightly-tests/max_versions.env:7]: Why don't we have the equivalent for https://github.com/solo-io/gloo/blob/main/.github/workflows/.env/pr-tests/versions.env? 

I know this is a bit outside the scope of this PR (and I'm happy if we do it in a separate PR), but given that we identified this drift, could we add the k8sgateway_api_version to the pr-test env file?

File: changelog/v1.19.0-beta2/issue_10456.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+changelog:
+  - type: FIX
+    issueLink: https://github.com/solo-io/gloo/issues/10456
+    resolvesIssue: true
```
Comment (sam-heilbron) [inline changelog/v1.19.0-beta2/issue_10456.yaml:4]: Is this true? We need to update the solo-projects dep to run this in CI as well before it is considered complete imo

File: .github/workflows/.env/nightly-tests/max_versions.env (hunk 1 above)
Comment (danehans) [reply .github/workflows/.env/nightly-tests/max_versions.env:7]: @sam-heilbron ^ is being resolved by https://github.com/solo-io/gloo/pull/10390.

File: changelog/v1.19.0-beta2/issue_10456.yaml (hunk 2 above)
Comment (danehans) [reply changelog/v1.19.0-beta2/issue_10456.yaml:4]: ^ issue is for gloo. IMHO solo-projects should have a separate issue to bump the gloo dep. I created https://github.com/solo-io/solo-projects/issues/7404 for this purpose.

**Comment by danehans** [comment]
Note that this task will be automated in the future with `CONFORMANCE_VERSION=v1.2.1 make sync-gateway-api` or `CONFORMANCE_VERSION=v1.2.1 make generated-code` when https://github.com/solo-io/gloo/pull/10446 merges.

**Pull request merged**